pkgs/development/tools/misc/texlab/default.nix

```diff
@@ -15,16 +15,18 @@ let
 in
 rustPlatform.buildRustPackage rec {
   pname = "texlab";
-  version = "5.9.0";
+  version = "5.9.2";
 
   src = fetchFromGitHub {
     owner = "latex-lsp";
     repo = "texlab";
     rev = "refs/tags/v${version}";
-    hash = "sha256-mF0vbKdTACGeL1NZ57kzZANxjcHX92WtRJoU2Z4iJiM=";
+    hash = "sha256-ZWvxi000wxjCzAe8PnzLb3z7smBc95gky0WyrkzVmEc=";
   };
 
-  cargoHash = "sha256-io0dkKDNoF+1n7D1q1JShd/Gj5Ce1Uh3dnM6uiTw0Vs=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
Oh oops sorry should've explained this more. This is using upstream's Cargo.lock. There's no functional difference, but uses `importCargoLock` to more finely split cargo dependencies into separate derivations: https://github.com/NixOS/nixpkgs/pull/217084

It's not really needed - I mainly just wanted to keep the same approach for all the rust packages I'm a maintainer on since #221716.

Reply:
I don't understand. The docs added in #221716 say:

> If the application has cargo `git` dependencies, the `cargoHash`/`cargoSha256`
approach will not work

Is this relevant for our case? I don't understand what would be the motivation to manually copy `Cargo.lock` from the upstream source if that is not the case - seems like a lot of headache for no reason.

Reply:
https://github.com/NixOS/nixpkgs/pull/221716 was just an approach to solve the core problems in https://github.com/NixOS/nixpkgs/pull/217084 with minimal impact, but https://github.com/NixOS/nixpkgs/pull/217084 also made some pretty good points for why `importCargoLock` should be used in general.

It's not a hard requirement in our case, but they way `importCargoLock` splits the dependencies into multiple derivations allows them to be more efficiently cached with other rust packages. For me I feel like that's worth the maintenance (especially since all other rust packages I maintain have a hard requirement for it) - but I can switch back to the hash approach if you'd prefer it instead.

Reply:
I am sorry, some of the terms in https://github.com/NixOS/nixpkgs/pull/217084 were not clear to me but I do understand the general motivation to vendor `Cargo.lock`. However, the main advantage of Nix compared to other distributions, of automatic updates, using `nixpkgs-update` or `nix-udpate` don't yet support updating these `Cargo.lock` files automatically for as far as I know. Hence I honestly tend to not adopt this change as of now, unless we get an alternative 1 command update method.